File: app/src/main/java/com/future/awaker/imageloader/ImageLoader.java
package com.future.awaker.imageloader;

import com.future.awaker.imageloader.glide.GlideImpl;

/**
 * Copyright ©2017 by ruzhan
 */

public final class ImageLoader {

    private static IImageLoader imageLoader;

    private ImageLoader() {
    }

    public static IImageLoader get() {
        if (imageLoader == null) {
            synchronized (ImageLoader.class) {
                if (imageLoader == null) {
                    imageLoader = new GlideImpl();
                }
            }
        }
        return imageLoader;
    }
}

File: common/src/main/java/com/ruzhan/common/imageloader/ImageLoader.kt
package com.ruzhan.common.imageloader

import com.ruzhan.common.imageloader.glide.GlideImpl

object ImageLoader {

    private var imageLoader: IImageLoader? = null

    fun get(): IImageLoader {
        if (imageLoader == null) {
            synchronized(ImageLoader::class.java) {
                if (imageLoader == null) {
                    imageLoader = GlideImpl()
                }
            }
        }
        return imageLoader!!
    }
}
